liblustreapi/src/lib.rs

```diff
@@ -265,29 +313,83 @@ impl Llapi {
         Ok(())
     }
 
-    pub fn rmfids(
+    fn llapi_rmfid(
         &self,
         mntpt: &str,
-        fidlist: impl IntoIterator<Item = String>,
-    ) -> Result<(), LiblustreError> {
-        // @@TODO replace with sys::llapi_rmfid once LU-12090 lands
+        fidlist: &Vec<String>,
+    ) -> Result<(), ()> {
+        let func: lib::Symbol<extern "C" fn(*const libc::c_char, *const RmfidsArg) -> libc::c_int> =
+            match unsafe { self.lib.get(b"llapi_rmfid\0") } {
+                Err(e) => {
+                    log::info!("Failed load llapi_rmfid: {:?}", e);
+                    return Err(());
+                }
+                Ok(f) => f,
+            };
+
+        let devptr = match CString::new(mntpt) {
+            Ok(c) => c.into_raw(),
+            Err(_) => return Err(()),
+        };
+
+        let mut fids: Vec<RmfidsArg> = Vec::with_capacity(fidlist.len() + 1);
+        fids.push(FidArrayHdr::new(0).into());
         for fidstr in fidlist {
-            self.rmfid(&mntpt, &fidstr).unwrap_or_else(|x| {
-                log::error!(
-                    "Couldn't remove fid {} with mountpoint {}: {}.",
-                    fidstr,
-                    mntpt,
-                    x
-                )
-            });
+            if let Ok(fid) = Fid::from_str(&fidstr) {
+                fids.push(fid.into());
+            } else {
+                log::error!("Bad fid format: {}", fidstr);
+            }
+        }
+        unsafe {
+            fids[0].hdr.nr = fids.len() as u32 - 1;
         }
 
-        Ok(())
+        let rc = func(devptr, fids.as_ptr());
+
+        // Ensure CString is freed
+        unsafe {
+            let _ = CString::from_raw(devptr);
+        }
+        if rc == 0 {
+            Ok(())
+        } else {
+            log::info!(
+                "Call to llapi_rmfids({}, [{}, ...]) failed: {}",
+                &mntpt,
+                unsafe { fids[0].hdr.nr },
+                rc
+            );
+            Err(())
```

Comment:
Should we return information in the Err object?

Reply:
No, because we ignore the error and use unlink instead.